lua/toggleterm/terminal.lua

```diff
@@ -179,7 +179,7 @@ end
 
 ---@private
 function Terminal:__spawn()
-  local cmd = self.cmd or vim.o.shell
+  local cmd = self.cmd or vim.g.defaultShell or vim.o.shell
```

Comment:
In able to this you'd need to
```lua
-- In config.lua add
config = {
  shell = vim.o.shell
}

-- In terminal.lua
local shell = config.get('shell')
local cmd = self.cmd or shell 
```

Reply:
I have put the setting in the global config like you said :)